rpc/legacyrpc/methods.go

```diff
@@ -1186,11 +1291,28 @@ func getMasterPubkey(icmd interface{}, w *wallet.Wallet) (interface{}, error) {
 
 // getStakeInfo gets a large amounts of information about the stake environment
 // and a number of statistics about local staking in the wallet.
-func getStakeInfo(icmd interface{}, w *wallet.Wallet, chainClient *dcrrpcclient.Client) (interface{}, error) {
+func getStakeInfo(s *Server, icmd interface{}) (interface{}, error) {
+	w, ok := s.walletLoader.LoadedWallet()
+	if !ok {
+		return nil, ErrUnloadedWallet
+	}
+
+	chainClient, ok := s.requireChainClient()
+	if !ok {
+		return nil, &dcrjson.RPCError{
+			Code:    -1,
+			Message: "Chain RPC is inactive",
+		}
+	}
+
+	rpcClient, err := chainClient.POSTClient()
+	if err != nil {
+		return nil, err
+	}
+
 	// Asynchronously query for the stake difficulty.
 	sdiffFuture := chainClient.GetStakeDifficultyAsync()
-
-	stakeInfo, err := w.StakeInfo(chainClient)
+	stakeInfo, err := w.StakeInfo(rpcClient)
```

Comment:
We shouldn't need to use the HTTP POST client here, the websocket RPC client is fine.